File: mobile/src/main/java/com/crow/stand_reminder/tool/WearTools.java
package com.crow.stand_reminder.tool;

import android.content.Context;

import androidx.annotation.WorkerThread;

import com.google.android.gms.tasks.Task;
import com.google.android.gms.tasks.Tasks;
import com.google.android.gms.wearable.Node;
import com.google.android.gms.wearable.Wearable;

import java.util.List;
import java.util.concurrent.ExecutionException;

public class WearTools
{
	@WorkerThread
	public static List<Node> getNodes(Context context)
	{
		Task<List<Node>> task = Wearable.getNodeClient(context).getConnectedNodes();

		try
		{
			return Tasks.await(task);
		}
		catch (ExecutionException | InterruptedException e)
		{
			e.printStackTrace();
			return null;
		}
	}
}

File: mobile/src/main/java/com/crow/stand_reminder/tool/WearTools.kt
package com.crow.stand_reminder.tool

import android.content.Context
import android.util.Log
import androidx.annotation.WorkerThread
import com.google.android.gms.tasks.Tasks
import com.google.android.gms.wearable.Node
import com.google.android.gms.wearable.Wearable
import java.util.concurrent.ExecutionException

object WearTools
{
	@WorkerThread
	fun getNodes(context: Context): List<Node>?
	{
		val task = Wearable.getNodeClient(context).connectedNodes

		return try
		{
			Tasks.await(task)
		}
		catch (e: ExecutionException)
		{
			Log.w("WEAR", e.message ?: "ExecutionException")
			null
		}
		catch (e: InterruptedException)
		{
			e.printStackTrace()
			null
		}
	}
}

